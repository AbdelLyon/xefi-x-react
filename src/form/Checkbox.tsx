import { forwardRef } from "react";
import type { CheckboxGroupProps, CheckboxProps } from "@heroui/react";
import { CheckboxGroup as CheckboxGroupRoot, Checkbox } from "@heroui/react";
import { mergeTailwindClasses } from "@/utils/utils";

type CheckboxItemProps = {
  label?: React.ReactNode;
} & Omit<CheckboxProps, "children">;

type CheckboxWrapperProps = {
  items: CheckboxItemProps[];
  groupClasses?: {
    base?: string;
    label?: string;
  };
  itemClasses?: {
    base?: string;
    label?: string;
    wrapper?: string;
  };
} & Omit<CheckboxGroupProps, "children">;

const CheckboxGroup = forwardRef<HTMLDivElement, CheckboxWrapperProps>(
  (
    {
      items,
      groupClasses,
      itemClasses,
      label = "Select options",
      defaultValue,
      ...props
    },
    ref,
  ): React.ReactNode => {
    const defaultGroupClasses = {
      base: "w-full",
      label: "text-medium font-semibold",
    };

    const defaultItemClasses = {
      base: "w-full",
      label: "text-small",
      wrapper: "",
    };

    return (
      <CheckboxGroupRoot
        ref={ref}
        label={label}
        defaultValue={defaultValue}
        {...props}
        classNames={{
          base: mergeTailwindClasses(
            defaultGroupClasses.base,
            groupClasses?.base,
          ),
          label: mergeTailwindClasses(
            defaultGroupClasses.label,
            groupClasses?.label,
          ),
        }}
      >
        {items.map(
          (item): React.ReactNode => (
            <Checkbox
              key={item.value}
              {...item}
              classNames={{
                base: mergeTailwindClasses(
                  defaultItemClasses.base,
                  itemClasses?.base,
                  item.className,
                ),
                label: mergeTailwindClasses(
                  defaultItemClasses.label,
                  itemClasses?.label,
                  item.classNames?.label,
                ),
                wrapper: mergeTailwindClasses(
                  defaultItemClasses.wrapper,
                  itemClasses?.wrapper,
                  item.classNames?.wrapper,
                ),
              }}
            >
              {item.label}
            </Checkbox>
          ),
        )}
      </CheckboxGroupRoot>
    );
  },
);

CheckboxGroup.displayName = "CheckboxGroup";

export { Checkbox, CheckboxGroup };
