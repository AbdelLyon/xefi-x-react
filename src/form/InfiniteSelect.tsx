import { useInfiniteScroll } from "@/hooks";
import type { SelectProps } from "@heroui/react";
import { Select, SelectItem } from "@heroui/react";
import type { JSX } from "react";
import { useState, useMemo } from "react";
import { mergeTailwindClasses } from "@/utils";

interface ProcessedItem<T> {
  data: T;
  uniqueIndex: number;
}

interface BaseInfiniteSelectProps<T>
  extends Omit<SelectProps, "items" | "children"> {
  renderItem: (item: T) => React.ReactNode;
  getItemKey: (item: T, index?: number) => string | number;
  className?: string;
  classNames?: {
    base?: string;
    trigger?: string;
    value?: string;
    popoverContent?: string;
    selectItem?: string;
  };
}

interface FetchFunctionProps<T> extends BaseInfiniteSelectProps<T> {
  fetchFunction: (
    offset: number,
    limit: number,
  ) => Promise<{
    items: T[];
    hasMore: boolean;
  }>;
  fetchDelay?: number;
  limit?: number;

  items?: undefined;
  hasMore?: undefined;
  isLoadingMore?: undefined;
  onLoadMore?: undefined;
}

interface ExternalDataProps<T> extends BaseInfiniteSelectProps<T> {
  items: T[];
  hasMore?: boolean;
  isLoadingMore?: boolean;
  onLoadMore?: () => void;

  fetchFunction?: undefined;
  fetchDelay?: undefined;
  limit?: undefined;
}

type InfiniteSelectProps<T> = FetchFunctionProps<T> | ExternalDataProps<T>;

const defaultClassNames = {
  base: "max-w-xs",
  trigger:
    "h-10 bg-white dark:bg-background data-[focus-visible=true]:outline-0 data-[focus=true]:border-outline data-[hover=true]:bg-transparant data-[hover=true]:border-outline",
  value: "text-small",
  popoverContent: "bg-white dark:bg-background",
  selectItem: "text-small",
} as const;

export function InfiniteSelect<T extends object>({
  fetchFunction: _fetchFunction,
  fetchDelay: _fetchDelay = 0,
  limit: _limit = 10,

  items: externalItems,
  hasMore: externalHasMore,
  isLoadingMore: externalIsLoading,
  onLoadMore: externalLoadMore,

  className,
  classNames,
  renderItem,
  getItemKey,
  selectionMode = "single",
  ...selectProps
}: InfiniteSelectProps<T>): JSX.Element {
  const [isOpen, setIsOpen] = useState(false);

  const processedItems = useMemo<ProcessedItem<T>[]>(() => {
    if (!externalItems) {
      return [];
    }

    return externalItems.map((item, index) => ({
      data: item,
      uniqueIndex: index,
    }));
  }, [externalItems]);

  const { scrollContainerRef } = useInfiniteScroll({
    hasMore: !!externalHasMore,
    isEnabled: isOpen,
    shouldUseLoader: false,
    onLoadMore: externalLoadMore,
  });

  const mergedClassNames = {
    base: mergeTailwindClasses(defaultClassNames.base, classNames?.base),
    trigger: mergeTailwindClasses(
      defaultClassNames.trigger,
      classNames?.trigger,
    ),
    value: mergeTailwindClasses(defaultClassNames.value, classNames?.value),
    popoverContent: mergeTailwindClasses(
      defaultClassNames.popoverContent,
      classNames?.popoverContent,
    ),
  };

  const rootClassName = mergeTailwindClasses(mergedClassNames.base, className);

  const selectItemClassName = mergeTailwindClasses(
    defaultClassNames.selectItem,
    classNames?.selectItem,
  );

  return (
    <Select
      className={rootClassName}
      classNames={mergedClassNames}
      isLoading={!!externalIsLoading}
      items={processedItems}
      scrollRef={scrollContainerRef}
      selectionMode={selectionMode}
      onOpenChange={(open): void => {
        setIsOpen(open);
        selectProps.onOpenChange?.(open);
      }}
      {...selectProps}
    >
      {(wrapper: ProcessedItem<T>): JSX.Element => (
        <SelectItem
          key={`${getItemKey(wrapper.data)}-${wrapper.uniqueIndex}`}
          textValue={String(getItemKey(wrapper.data))}
          className={selectItemClassName}
        >
          {renderItem(wrapper.data)}
        </SelectItem>
      )}
    </Select>
  );
}
