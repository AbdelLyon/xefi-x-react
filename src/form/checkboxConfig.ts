/**
 * Checkbox component configuration
 * Provides standardized styling and behavior patterns
 */

import { createClassNamesConfig } from "@/utils/classNames";
import { baseFormFieldClasses, type FormFieldConfig, type FormGroupConfig } from "./formConfig";
import type { ReactNode } from "react";

/**
 * Default class names for checkbox components
 */
export const defaultCheckboxClassNames = {
  ...baseFormFieldClasses,
  base: "group relative max-w-fit inline-flex items-center justify-start cursor-pointer",
  wrapper: "relative inline-flex items-center justify-center flex-shrink-0",
  icon: "z-10 w-4 h-4 opacity-0 group-data-[selected=true]:opacity-100 transition-opacity",
  label: "ml-2 text-foreground select-none",
} as const;

/**
 * Checkbox size class configurations
 */
export const checkboxSizeClasses = {
  sm: {
    wrapper: "w-4 h-4 mr-2",
    icon: "w-3 h-3",
    label: "text-sm",
  },
  md: {
    wrapper: "w-5 h-5 mr-2",
    icon: "w-3 h-3",
    label: "text-base",
  },
  lg: {
    wrapper: "w-6 h-6 mr-3",
    icon: "w-4 h-4",
    label: "text-lg",
  },
} as const;

/**
 * Checkbox color class configurations
 */
export const checkboxColorClasses = {
  default: {
    wrapper: "border-default-300 data-[selected=true]:border-default data-[selected=true]:bg-default",
    icon: "text-default-foreground",
  },
  primary: {
    wrapper: "border-default-300 data-[selected=true]:border-primary data-[selected=true]:bg-primary",
    icon: "text-primary-foreground",
  },
  secondary: {
    wrapper: "border-default-300 data-[selected=true]:border-secondary data-[selected=true]:bg-secondary",
    icon: "text-secondary-foreground",
  },
  success: {
    wrapper: "border-default-300 data-[selected=true]:border-success data-[selected=true]:bg-success",
    icon: "text-success-foreground",
  },
  warning: {
    wrapper: "border-default-300 data-[selected=true]:border-warning data-[selected=true]:bg-warning",
    icon: "text-warning-foreground",
  },
  danger: {
    wrapper: "border-default-300 data-[selected=true]:border-danger data-[selected=true]:bg-danger",
    icon: "text-danger-foreground",
  },
} as const;

/**
 * Checkbox state classes
 */
export const checkboxStateClasses = {
  default: {
    base: "",
    wrapper: "",
  },
  invalid: {
    base: "",
    wrapper: "border-danger data-[selected=true]:border-danger data-[selected=true]:bg-danger",
  },
  valid: {
    base: "",
    wrapper: "border-success data-[selected=true]:border-success data-[selected=true]:bg-success",
  },
  disabled: {
    base: "opacity-50 cursor-not-allowed",
    wrapper: "border-default-200 bg-default-100",
  },
} as const;

/**
 * Create checkbox class configuration
 */
export const checkboxClassConfig = createClassNamesConfig(defaultCheckboxClassNames);

/**
 * Get checkbox size classes
 */
export const getCheckboxSizeClasses = (size: keyof typeof checkboxSizeClasses = "md") => {
  return checkboxSizeClasses[size] || checkboxSizeClasses.md;
};

/**
 * Get checkbox color classes
 */
export const getCheckboxColorClasses = (color: keyof typeof checkboxColorClasses = "primary") => {
  return checkboxColorClasses[color] || checkboxColorClasses.primary;
};

/**
 * Get checkbox state classes
 */
export const getCheckboxStateClasses = (state: keyof typeof checkboxStateClasses = "default") => {
  return checkboxStateClasses[state] || checkboxStateClasses.default;
};

/**
 * Checkbox item configuration
 */
export interface CheckboxItemConfig extends FormFieldConfig {
  /** Checkbox value */
  value: string;
  /** Checkbox label content */
  label?: ReactNode;
  /** Whether checkbox is checked */
  checked?: boolean;
  /** Whether checkbox is indeterminate */
  indeterminate?: boolean;
  /** Custom icon for checked state */
  icon?: ReactNode;
  /** Custom color theme */
  color?: keyof typeof checkboxColorClasses;
  /** Change handler */
  onChange?: (checked: boolean) => void;
}

/**
 * Checkbox group configuration
 */
export interface CheckboxGroupConfig extends FormGroupConfig {
  /** Array of checkbox items */
  items: CheckboxItemConfig[];
  /** Current selected values */
  value?: string[];
  /** Default selected values */
  defaultValue?: string[];
  /** Selection change handler */
  onChange?: (values: string[]) => void;
  /** Maximum selections allowed */
  maxSelections?: number;
  /** Minimum selections required */
  minSelections?: number;
  /** Whether to show select all option */
  showSelectAll?: boolean;
  /** Select all label */
  selectAllLabel?: ReactNode;
}

/**
 * Validate checkbox group selection
 */
export const validateCheckboxGroup = (
  selectedValues: string[],
  config: CheckboxGroupConfig
): { valid: boolean; errors: string[]; } => {
  const errors: string[] = [];

  // Check minimum selections
  if (config.minSelections && selectedValues.length < config.minSelections) {
    errors.push(`Please select at least ${config.minSelections} option${config.minSelections > 1 ? "s" : ""}`);
  }

  // Check maximum selections
  if (config.maxSelections && selectedValues.length > config.maxSelections) {
    errors.push(`Please select no more than ${config.maxSelections} option${config.maxSelections > 1 ? "s" : ""}`);
  }

  // Check required validation
  if (config.required && selectedValues.length === 0) {
    errors.push("Please select at least one option");
  }

  return {
    valid: errors.length === 0,
    errors,
  };
};

/**
 * Create checkbox props with defaults
 */
export const createCheckboxProps = <T extends Record<string, unknown>>(
  userProps: T,
  config: CheckboxItemConfig
) => {
  const sizeClasses = getCheckboxSizeClasses(config.size);
  const colorClasses = getCheckboxColorClasses(config.color);
  const stateClasses = getCheckboxStateClasses(config.state === 'loading' ? 'default' : config.state);

  const mergedClasses = checkboxClassConfig.mergeClasses({
    ...sizeClasses,
    ...colorClasses,
    ...stateClasses,
    ...config.classNames,
  });

  return {
    value: config.value,
    isSelected: config.checked || false,
    isIndeterminate: config.indeterminate || false,
    isRequired: config.required || false,
    isDisabled: config.disabled || false,
    isInvalid: config.state === "invalid",
    size: config.size || "md",
    color: config.color || "primary",
    classNames: mergedClasses,
    children: config.label,
    icon: config.icon,
    onValueChange: config.onChange,
    ...userProps,
  };
};

/**
 * Default checkbox behavior
 */
export const defaultCheckboxBehavior = {
  size: "md" as const,
  color: "primary" as const,
  state: "default" as const,
  required: false,
  disabled: false,
  checked: false,
  indeterminate: false,
} as const;