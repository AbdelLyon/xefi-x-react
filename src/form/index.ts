import "@/index.css";
export { RadioGroup } from "@/form/Radio";
export { Checkbox, CheckboxGroup } from "@/form/Checkbox";
export { InputOtp } from "@/form/InputOpt";
export { Input } from "@/form/Input";
export { Textarea } from "@/form/Textarea";
export { Switch } from "@/form/Switch";
export { Select, type SelectOption } from "@/form/Select";
export { InfiniteSelect } from "@/form/InfiniteSelect";
export { InfiniteAutocomplete } from "@/form/InfiniteAutocomplete";
