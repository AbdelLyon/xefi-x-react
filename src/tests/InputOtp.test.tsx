import { InputOtp } from "@/form";
import { render, screen, fireEvent } from "@testing-library/react";
import { describe, it, expect } from "vitest";

describe("Composant InputOtp", (): void => {
  describe("Snapshots", (): void => {
    it("devrait correspondre au snapshot avec les props par défaut", (): void => {
      const { container } = render(<InputOtp />);
      expect(container).toMatchSnapshot();
    });

    it("devrait correspondre au snapshot avec une longueur personnalisée", (): void => {
      const { container } = render(<InputOtp length={4} />);
      expect(container).toMatchSnapshot();
    });

    it("devrait correspondre au snapshot avec un label", (): void => {
      const { container } = render(<InputOtp label="Code OTP" />);
      expect(container).toMatchSnapshot();
    });

    it("devrait correspondre au snapshot avec des classes personnalisées", (): void => {
      const { container } = render(
        <InputOtp
          labelClasses="custom-label"
          containerClasses="custom-container"
        />,
      );
      expect(container).toMatchSnapshot();
    });
  });

  describe("Rendu", (): void => {
    it("devrait rendre le nombre correct de champs de saisie", (): void => {
      render(<InputOtp length={4} />);
      const input = screen.getByRole("textbox");
      expect(input).toHaveAttribute("maxlength", "4");
    });

    it("devrait rendre le label quand il est fourni", (): void => {
      render(<InputOtp label="Code OTP" />);
      expect(screen.getByText("Code OTP")).toBeInTheDocument();
    });
  });

  describe("Comportement", (): void => {
    it("devrait mettre à jour la valeur lors de la saisie", (): void => {
      render(<InputOtp />);
      const input = screen.getByRole("textbox");
      fireEvent.change(input, { target: { value: "1" } });
      expect(input).toHaveValue("1");
    });

    it("devrait limiter la saisie à la longueur spécifiée", (): void => {
      render(<InputOtp length={4} />);
      const input = screen.getByRole("textbox");
      fireEvent.change(input, { target: { value: "12345" } });
      expect(input).toHaveValue("1234");
    });
  });

  describe("Accessibilité", (): void => {
    it("devrait avoir des attributs aria appropriés", (): void => {
      render(<InputOtp />);
      const container = screen.getByRole("base");
      expect(container).toHaveAttribute(
        "aria-label",
        "One-time password input",
      );
    });

    it("devrait avoir un input de type numérique", (): void => {
      render(<InputOtp />);
      const input = screen.getByRole("textbox");
      expect(input).toHaveAttribute("inputmode", "numeric");
    });
  });
});
