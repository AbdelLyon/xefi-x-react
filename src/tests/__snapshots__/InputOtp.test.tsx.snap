// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Composant InputOtp > Snapshots > devrait correspondre au snapshot avec des classes personnalisées 1`] = `
<div>
  <div
    class="flex flex-col custom-container"
  >
    <div
      aria-label="One-time password input"
      class="relative flex flex-col w-fit"
      data-slot="base"
      role="base"
    >
      <noscript />
      <div
        class="group flex items-center has-[:disabled]:opacity-60"
        data-input-otp-container="true"
        style="position: relative; cursor: text; user-select: none; pointer-events: none; --root-height: 0px;"
      >
        <div
          aria-label="One-time password input"
          class="inline-flex gap-x-1 py-2"
          data-slot="segment-wrapper"
        >
          <div
            class="font-semibold flex justify-center items-center data-[active=true]:border-default-400 data-[active=true]:scale-110 shadow-sm hover:bg-danger outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 border-transparent rounded-md h-10 min-h-10 w-10 min-w-10 text-small transition duration-150 bg-default-100 data-[active=true]:bg-default-200"
            data-slot="segment2"
            role="presentation"
          >
            <div />
          </div>
          <div
            class="font-semibold flex justify-center items-center data-[active=true]:border-default-400 data-[active=true]:scale-110 shadow-sm hover:bg-danger outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 border-transparent rounded-md h-10 min-h-10 w-10 min-w-10 text-small transition duration-150 bg-default-100 data-[active=true]:bg-default-200"
            data-slot="segment2"
            role="presentation"
          >
            <div />
          </div>
          <div
            class="font-semibold flex justify-center items-center data-[active=true]:border-default-400 data-[active=true]:scale-110 shadow-sm hover:bg-danger outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 border-transparent rounded-md h-10 min-h-10 w-10 min-w-10 text-small transition duration-150 bg-default-100 data-[active=true]:bg-default-200"
            data-slot="segment2"
            role="presentation"
          >
            <div />
          </div>
          <div
            class="font-semibold flex justify-center items-center data-[active=true]:border-default-400 data-[active=true]:scale-110 shadow-sm hover:bg-danger outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 border-transparent rounded-md h-10 min-h-10 w-10 min-w-10 text-small transition duration-150 bg-default-100 data-[active=true]:bg-default-200"
            data-slot="segment2"
            role="presentation"
          >
            <div />
          </div>
          <div
            class="font-semibold flex justify-center items-center data-[active=true]:border-default-400 data-[active=true]:scale-110 shadow-sm hover:bg-danger outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 border-transparent rounded-md h-10 min-h-10 w-10 min-w-10 text-small transition duration-150 bg-default-100 data-[active=true]:bg-default-200"
            data-slot="segment2"
            role="presentation"
          >
            <div />
          </div>
          <div
            class="font-semibold flex justify-center items-center data-[active=true]:border-default-400 data-[active=true]:scale-110 shadow-sm hover:bg-danger outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 border-transparent rounded-md h-10 min-h-10 w-10 min-w-10 text-small transition duration-150 bg-default-100 data-[active=true]:bg-default-200"
            data-slot="segment2"
            role="presentation"
          >
            <div />
          </div>
        </div>
        <div
          style="position: absolute; inset: 0; pointer-events: none;"
        >
          <input
            autocomplete="one-time-code"
            data-input-otp="true"
            data-input-otp-placeholder-shown="true"
            inputmode="numeric"
            maxlength="6"
            minlength="6"
            pattern="^[0-9]*$"
            style="position: absolute; inset: 0; width: 100%; height: 100%; display: flex; text-align: center; opacity: 1; color: transparent; pointer-events: all; background: transparent; caret-color: transparent; border: 0px solid transparent; outline: 0 solid transparent; box-shadow: none; line-height: 1; letter-spacing: -.5em; font-size: var(--root-height); font-family: monospace; font-variant-numeric: tabular-nums;"
            title=""
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Composant InputOtp > Snapshots > devrait correspondre au snapshot avec les props par défaut 1`] = `
<div>
  <div
    class="flex flex-col"
  >
    <div
      aria-label="One-time password input"
      class="relative flex flex-col w-fit"
      data-slot="base"
      role="base"
    >
      <noscript />
      <div
        class="group flex items-center has-[:disabled]:opacity-60"
        data-input-otp-container="true"
        style="position: relative; cursor: text; user-select: none; pointer-events: none; --root-height: 0px;"
      >
        <div
          aria-label="One-time password input"
          class="inline-flex gap-x-1 py-2"
          data-slot="segment-wrapper"
        >
          <div
            class="font-semibold flex justify-center items-center data-[active=true]:border-default-400 data-[active=true]:scale-110 shadow-sm hover:bg-danger outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 border-transparent rounded-md h-10 min-h-10 w-10 min-w-10 text-small transition duration-150 bg-default-100 data-[active=true]:bg-default-200"
            data-slot="segment2"
            role="presentation"
          >
            <div />
          </div>
          <div
            class="font-semibold flex justify-center items-center data-[active=true]:border-default-400 data-[active=true]:scale-110 shadow-sm hover:bg-danger outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 border-transparent rounded-md h-10 min-h-10 w-10 min-w-10 text-small transition duration-150 bg-default-100 data-[active=true]:bg-default-200"
            data-slot="segment2"
            role="presentation"
          >
            <div />
          </div>
          <div
            class="font-semibold flex justify-center items-center data-[active=true]:border-default-400 data-[active=true]:scale-110 shadow-sm hover:bg-danger outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 border-transparent rounded-md h-10 min-h-10 w-10 min-w-10 text-small transition duration-150 bg-default-100 data-[active=true]:bg-default-200"
            data-slot="segment2"
            role="presentation"
          >
            <div />
          </div>
          <div
            class="font-semibold flex justify-center items-center data-[active=true]:border-default-400 data-[active=true]:scale-110 shadow-sm hover:bg-danger outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 border-transparent rounded-md h-10 min-h-10 w-10 min-w-10 text-small transition duration-150 bg-default-100 data-[active=true]:bg-default-200"
            data-slot="segment2"
            role="presentation"
          >
            <div />
          </div>
          <div
            class="font-semibold flex justify-center items-center data-[active=true]:border-default-400 data-[active=true]:scale-110 shadow-sm hover:bg-danger outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 border-transparent rounded-md h-10 min-h-10 w-10 min-w-10 text-small transition duration-150 bg-default-100 data-[active=true]:bg-default-200"
            data-slot="segment2"
            role="presentation"
          >
            <div />
          </div>
          <div
            class="font-semibold flex justify-center items-center data-[active=true]:border-default-400 data-[active=true]:scale-110 shadow-sm hover:bg-danger outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 border-transparent rounded-md h-10 min-h-10 w-10 min-w-10 text-small transition duration-150 bg-default-100 data-[active=true]:bg-default-200"
            data-slot="segment2"
            role="presentation"
          >
            <div />
          </div>
        </div>
        <div
          style="position: absolute; inset: 0; pointer-events: none;"
        >
          <input
            autocomplete="one-time-code"
            data-input-otp="true"
            data-input-otp-placeholder-shown="true"
            inputmode="numeric"
            maxlength="6"
            minlength="6"
            pattern="^[0-9]*$"
            style="position: absolute; inset: 0; width: 100%; height: 100%; display: flex; text-align: center; opacity: 1; color: transparent; pointer-events: all; background: transparent; caret-color: transparent; border: 0px solid transparent; outline: 0 solid transparent; box-shadow: none; line-height: 1; letter-spacing: -.5em; font-size: var(--root-height); font-family: monospace; font-variant-numeric: tabular-nums;"
            title=""
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Composant InputOtp > Snapshots > devrait correspondre au snapshot avec un label 1`] = `
<div>
  <div
    class="flex flex-col"
  >
    <p
      class="text-small mb-2"
    >
      Code OTP
    </p>
    <div
      aria-label="One-time password input"
      class="relative flex flex-col w-fit"
      data-slot="base"
      role="base"
    >
      <noscript />
      <div
        class="group flex items-center has-[:disabled]:opacity-60"
        data-input-otp-container="true"
        style="position: relative; cursor: text; user-select: none; pointer-events: none; --root-height: 0px;"
      >
        <div
          aria-label="One-time password input"
          class="inline-flex gap-x-1 py-2"
          data-slot="segment-wrapper"
        >
          <div
            class="font-semibold flex justify-center items-center data-[active=true]:border-default-400 data-[active=true]:scale-110 shadow-sm hover:bg-danger outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 border-transparent rounded-md h-10 min-h-10 w-10 min-w-10 text-small transition duration-150 bg-default-100 data-[active=true]:bg-default-200"
            data-slot="segment2"
            role="presentation"
          >
            <div />
          </div>
          <div
            class="font-semibold flex justify-center items-center data-[active=true]:border-default-400 data-[active=true]:scale-110 shadow-sm hover:bg-danger outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 border-transparent rounded-md h-10 min-h-10 w-10 min-w-10 text-small transition duration-150 bg-default-100 data-[active=true]:bg-default-200"
            data-slot="segment2"
            role="presentation"
          >
            <div />
          </div>
          <div
            class="font-semibold flex justify-center items-center data-[active=true]:border-default-400 data-[active=true]:scale-110 shadow-sm hover:bg-danger outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 border-transparent rounded-md h-10 min-h-10 w-10 min-w-10 text-small transition duration-150 bg-default-100 data-[active=true]:bg-default-200"
            data-slot="segment2"
            role="presentation"
          >
            <div />
          </div>
          <div
            class="font-semibold flex justify-center items-center data-[active=true]:border-default-400 data-[active=true]:scale-110 shadow-sm hover:bg-danger outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 border-transparent rounded-md h-10 min-h-10 w-10 min-w-10 text-small transition duration-150 bg-default-100 data-[active=true]:bg-default-200"
            data-slot="segment2"
            role="presentation"
          >
            <div />
          </div>
          <div
            class="font-semibold flex justify-center items-center data-[active=true]:border-default-400 data-[active=true]:scale-110 shadow-sm hover:bg-danger outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 border-transparent rounded-md h-10 min-h-10 w-10 min-w-10 text-small transition duration-150 bg-default-100 data-[active=true]:bg-default-200"
            data-slot="segment2"
            role="presentation"
          >
            <div />
          </div>
          <div
            class="font-semibold flex justify-center items-center data-[active=true]:border-default-400 data-[active=true]:scale-110 shadow-sm hover:bg-danger outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 border-transparent rounded-md h-10 min-h-10 w-10 min-w-10 text-small transition duration-150 bg-default-100 data-[active=true]:bg-default-200"
            data-slot="segment2"
            role="presentation"
          >
            <div />
          </div>
        </div>
        <div
          style="position: absolute; inset: 0; pointer-events: none;"
        >
          <input
            autocomplete="one-time-code"
            data-input-otp="true"
            data-input-otp-placeholder-shown="true"
            inputmode="numeric"
            maxlength="6"
            minlength="6"
            pattern="^[0-9]*$"
            style="position: absolute; inset: 0; width: 100%; height: 100%; display: flex; text-align: center; opacity: 1; color: transparent; pointer-events: all; background: transparent; caret-color: transparent; border: 0px solid transparent; outline: 0 solid transparent; box-shadow: none; line-height: 1; letter-spacing: -.5em; font-size: var(--root-height); font-family: monospace; font-variant-numeric: tabular-nums;"
            title=""
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Composant InputOtp > Snapshots > devrait correspondre au snapshot avec une longueur personnalisée 1`] = `
<div>
  <div
    class="flex flex-col"
  >
    <div
      aria-label="One-time password input"
      class="relative flex flex-col w-fit"
      data-slot="base"
      role="base"
    >
      <noscript />
      <div
        class="group flex items-center has-[:disabled]:opacity-60"
        data-input-otp-container="true"
        style="position: relative; cursor: text; user-select: none; pointer-events: none; --root-height: 0px;"
      >
        <div
          aria-label="One-time password input"
          class="inline-flex gap-x-1 py-2"
          data-slot="segment-wrapper"
        >
          <div
            class="font-semibold flex justify-center items-center data-[active=true]:border-default-400 data-[active=true]:scale-110 shadow-sm hover:bg-danger outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 border-transparent rounded-md h-10 min-h-10 w-10 min-w-10 text-small transition duration-150 bg-default-100 data-[active=true]:bg-default-200"
            data-slot="segment2"
            role="presentation"
          >
            <div />
          </div>
          <div
            class="font-semibold flex justify-center items-center data-[active=true]:border-default-400 data-[active=true]:scale-110 shadow-sm hover:bg-danger outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 border-transparent rounded-md h-10 min-h-10 w-10 min-w-10 text-small transition duration-150 bg-default-100 data-[active=true]:bg-default-200"
            data-slot="segment2"
            role="presentation"
          >
            <div />
          </div>
          <div
            class="font-semibold flex justify-center items-center data-[active=true]:border-default-400 data-[active=true]:scale-110 shadow-sm hover:bg-danger outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 border-transparent rounded-md h-10 min-h-10 w-10 min-w-10 text-small transition duration-150 bg-default-100 data-[active=true]:bg-default-200"
            data-slot="segment2"
            role="presentation"
          >
            <div />
          </div>
          <div
            class="font-semibold flex justify-center items-center data-[active=true]:border-default-400 data-[active=true]:scale-110 shadow-sm hover:bg-danger outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 border-transparent rounded-md h-10 min-h-10 w-10 min-w-10 text-small transition duration-150 bg-default-100 data-[active=true]:bg-default-200"
            data-slot="segment2"
            role="presentation"
          >
            <div />
          </div>
        </div>
        <div
          style="position: absolute; inset: 0; pointer-events: none;"
        >
          <input
            autocomplete="one-time-code"
            data-input-otp="true"
            data-input-otp-placeholder-shown="true"
            inputmode="numeric"
            maxlength="4"
            minlength="4"
            pattern="^[0-9]*$"
            style="position: absolute; inset: 0; width: 100%; height: 100%; display: flex; text-align: center; opacity: 1; color: transparent; pointer-events: all; background: transparent; caret-color: transparent; border: 0px solid transparent; outline: 0 solid transparent; box-shadow: none; line-height: 1; letter-spacing: -.5em; font-size: var(--root-height); font-family: monospace; font-variant-numeric: tabular-nums;"
            title=""
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
